src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/SelectResultSet.java

```diff
@@ -273,6 +275,22 @@ private String getFieldName(Field field) {
             case "min":
             case "max":
                 return Schema.Type.DOUBLE;
+            case "script": {
+                ScriptMethodField smf = (ScriptMethodField) field;
+                // TODO: this information is disconnected from the function definitions in SQLFunctions.
+                // Refactor SQLFunctions to have functions self-explanatory (types, scripts) and pluggable
+                // (similar to Strategy pattern).
+
+                switch (smf.getFunctionName()) {
+                    case "date_format": {
```

Comment:
I remember another immediate requirement is to support arithmetic expression, ex. "SELECT age / 2 FROM A".

For "date_format", you've aware of it, why still return TEXT? I thought we either maintain a mapping from known function to its return value type (like what you do here), or we return TEXT for everything for now.

Reply:
Changed in next version, will publish after all tests will pass